src/plugins/patch-binary.ts

```diff
@@ -0,0 +1,54 @@
+import fs from 'fs'
+import MagicString from 'magic-string'
+import path from 'path'
+import type { OutputChunk, Plugin, SourceMapInput } from 'rollup'
+
+export const patchBinary = (executablePaths: string[]): Plugin => ({
+  name: 'patchBinary',
+
+  renderChunk: (code, chunk, outputOptions) => {
```

Comment:
we can do the prepend in `transform`, where we check if the module id equals to the current build's entry of binary file, and decide to prepend the shebang